reporter/src/main/kotlin/reporters/AmazonOssAttributionBuilderReporter.kt

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import org.ossreviewtoolkit.model.jsonMapper
+import org.ossreviewtoolkit.model.utils.getDetectedLicensesWithCopyrights
+import org.ossreviewtoolkit.reporter.HTTP_CACHE_PATH
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.model.AmazonOssAttributionBuilderService
+import org.ossreviewtoolkit.utils.OkHttpClientHelper
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.showStackTrace
+
+import java.io.IOException
+import java.io.OutputStream
+import java.net.ConnectException
+import java.time.Instant
+
+import okhttp3.Credentials
+
+import retrofit2.Response
+
+private fun <T> getBodyOrHandleError(response: Response<T>, errorTitle: String): T {
+    val body = response.body()
+    if (response.isSuccessful && body != null) return body
+
+    val errorMessage = response.errorBody()?.let { errorBody ->
+        val errorResponse = jsonMapper.readValue(
+            errorBody.string(),
+            AmazonOssAttributionBuilderService.ErrorResponse::class.java
+        )
+
+        errorResponse.error
+    } ?: "Error code ${response.code()}"
+
+    throw IOException("$errorTitle: $errorMessage")
+}
+
+class AmazonOssAttributionBuilderReporter : Reporter {
```

Comment:
Please add KDoc for this class explaining what it does, especially what kind of output is generated.

Reply:
done